src/mflux/server/openai_server.py

```diff
@@ -0,0 +1,243 @@
+                lora_scales = [request.lora_scale]
+        
+        # Get the model using the model manager
+        flux, model_key = model_manager.get_model(
+            model_name=model,
+            base_model=request.base_model,
+            quantize=request.quantize,
+            local_path=request.local_path,
+            lora_paths=lora_paths,
+            lora_scales=lora_scales,
+            keep_alive=request.keep_alive
+        )
+        
+        # Register optional callbacks
+        if stepwise_dir:
+            handler = StepwiseHandler(flux=flux, output_dir=stepwise_dir)
+            CallbackRegistry.register_before_loop(handler)
+            CallbackRegistry.register_in_loop(handler)
+            CallbackRegistry.register_interrupt(handler)
+
+        memory_saver = None
+        if request.low_ram:
+            memory_saver = MemorySaver(flux)
+            CallbackRegistry.register_before_loop(memory_saver)
+            CallbackRegistry.register_in_loop(memory_saver)
+            CallbackRegistry.register_after_loop(memory_saver)
+        
+        images = []
+        
+        try:
+            for i in range(request.n or 1):
+                if request.seed is not None:
+                    current_seed = request.seed + i
+                else:
+                    current_seed = int.from_bytes(os.urandom(4), byteorder='big')
+                
+                # Generate the image
+                generated_image = flux.generate_image(
+                    seed=current_seed,
+                    prompt=request.prompt,
+                    config=Config(
+                        num_inference_steps=request.n or 2,
+                        height=height,
+                        width=width,
+                        guidance=request.guidance or 4.0,
+                        image_path=image_path,
+                        image_strength=request.image_strength,
+                        controlnet_strength=request.controlnet_strength
+                    )
+                )
+                
+                # Save image to buffer via temporary file
+                buffer = io.BytesIO()
+                temp_img_path = f"{create_temp_directory()}/temp_img_{i}.png"
+
+                # Save with or without metadata
+                generated_image.save(
+                    path=temp_img_path, 
+                    export_json_metadata=request.metadata or False
+                )
+
+                # Read the file back
+                with open(temp_img_path, "rb") as f:
+                    buffer.write(f.read())
+
+                # Clean up temp file
+                cleanup_temp_file(temp_img_path)
+
+                # Encode image to base64
+                buffer.seek(0)
+                b64_img = base64.b64encode(buffer.getvalue()).decode('utf-8')
+                
+                # Add the image to our response list
+                images.append(Image(b64_json=b64_img))
```

Comment:
Using `NamedTemporaryFile` something like this should be a bit easier.
```suggestion
                with tempfile.NamedTemporaryFile(prefix="mflux", suffix="png") as tf:
                    generated_image.save(
                        path=tf.name, 
                        export_json_metadata=request.metadata,
                    )
                    tf.seek(0)
                    b64_img = base64.b64encode(tf.read()).decode('utf-8')
                
                # Add the image to our response list
                images.append(Image(b64_json=b64_img))
```